packages/netlify-cms-backend-github/src/API.ts

```diff
@@ -870,145 +870,103 @@ export default class API {
         entry: { path: entry.path, sha: entry.sha },
         files: mediaFilesList,
       };
-      const updatedMetadata = { ...metadata, pr, title, description, objects };
 
-      if (options.hasAssetStore) {
-        await this.storeMetadata(contentKey, updatedMetadata);
-        return this.patchBranch(branchName, commit.sha);
-      }
-
-      if (pr) {
-        return this.rebasePullRequest(pr.number, branchName, contentKey, updatedMetadata, commit);
-      } else if (this.useOpenAuthoring) {
-        // if a PR hasn't been created yet for the forked repo, just patch the branch
-        await this.patchBranch(branchName, commit.sha, { force: true });
-      }
+      const updatedMetadata = { ...metadata, pr, title, description, objects };
 
-      return this.storeMetadata(contentKey, updatedMetadata);
+      await this.storeMetadata(contentKey, updatedMetadata);
+      return this.patchBranch(branchName, commit.sha, { force: true });
     }
   }
 
-  /**
-   * Rebase a pull request onto the latest HEAD of it's target base branch
-   * (should generally be the configured backend branch). Only rebases changes
-   * in the entry file.
-   */
-  async rebasePullRequest(
-    prNumber: number,
+  async compareBranchToDefault(
     branchName: string,
-    contentKey: string,
-    metadata: Metadata,
-    head: Commit,
-  ) {
-    const { path } = metadata.objects.entry;
+  ): Promise<{ baseCommit: GitHubCompareBaseCommit; commits: GitHubCompareCommits }> {
+    const headReference = await this.getHeadReference(branchName);
+    const { base_commit: baseCommit, commits }: GitHubCompareResponse = await this.request(
+      `${this.originRepoURL}/compare/${this.branch}...${headReference}`,
+    );
+    return { baseCommit, commits };
+  }
 
-    try {
-      /**
-       * Get the published branch and create new commits over it. If the pull
-       * request is up to date, no rebase will occur.
-       */
-      const [baseBranch, commits] = await Promise.all([
-        this.getBranch(),
-        this.getPullRequestCommits(prNumber),
-      ]);
+  async getCommitsDiff(baseSha: string, headSha: string): Promise<GitHubCompareFiles> {
+    const { files }: GitHubCompareResponse = await this.request(
+      `${this.repoURL}/compare/${baseSha}...${headSha}`,
+    );
+    return files;
+  }
+
+  async rebaseSingleCommit(baseCommit: GitHubCompareCommit, commit: GitHubCompareCommit) {
+    // first get the diff between the commits
+    const files = await this.getCommitsDiff(commit.parents[0].sha, commit.sha);
+    const treeFiles = files.reduce((arr, file) => {
+      if (file.status === 'removed') {
+        // delete the file
+        arr.push({ sha: null, path: file.filename });
```

Comment:
Not necessary for this PR, but we should prefer immutable operations over mutating ones.